### PR TITLE
Update Expanse build to use Charm 7 and Intel MPI

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: support/Environments/expanse_gcc.sh [hunk 1]
```diff
@@ -5,19 +5,36 @@
...
 # Load system modules
 spectre_load_sys_modules() {
+    # Make sure the core default modules are loaded.
+    # Without these you can get weird compilation errors
+    # that make no sense.
+    module load shared cpu sdsc DefaultModules slurm/expanse numactl
+
     module load gcc/10.2.0
-    module load intel-mkl/2019.1.144
+    module load openblas/0.3.10-openmp
```
Comment (knelli2) [inline support/Environments/expanse_gcc.sh:14]: Why `openblas` vs `intel-mkl`?

File: support/Environments/setup/expanse_gcc.sh [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+pip install pyyaml
+pip install matplotlib
+
+cd $dep_dir
+
+if [ -f $dep_dir/modules/spectre_zlib ]; then
+    echo "zlib is already installed"
+else
+    echo "Installing zlib module..."
+    # The zlib configuration is a bit questionable, so we set up a module.
```
Comment (knelli2) [inline support/Environments/setup/expanse_gcc.sh]: Why is it questionable?

**Review by knelli2 — COMMENTED** [review]
Haven't tried this yet, but had a couple questions.

File: support/Environments/expanse_gcc.sh (hunk 1 above)
Comment (nilsdeppe) [reply support/Environments/expanse_gcc.sh:14]: Intel MKL didn't build and link correctly for me

**Comment by nilsdeppe** [comment]
Rebased on develop

**Comment by geoffrey4444** [comment]
I am testing this branch on expanse...will let you know if I run into trouble

**Comment by geoffrey4444** [comment]
Following the instructions on the website, I installed a fresh copy of the modules:

```
export SPECTRE_HOME=${HOME}/Codes/spectre
source $SPECTRE_HOME/support/Environments/expanse_gcc.sh
export $SPECTRE_DEPS=${HOME}/apps
spectre_setup_modules
spectre_run_cmake
```

But when I now try to cmake, I get an error that python can't be found, even though the installed python is in my path

```
[geoffrey@login01 build_expansesupport]$ which python
~/apps/py_env/bin/python
[geoffrey@login01 build_expansesupport]$ python
Python 3.8.5 (default, Oct  7 2020, 14:21:11) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 
[geoffrey@login01 build_expansesupport]$ spectre_run_cmake 
-- SpECTRE release version: 2022.05.05
-- The CXX compiler identification is GNU 10.2.0
-- The C compiler identification is GNU 10.2.0
-- The Fortran compiler identification is GNU 10.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-n7su7jf54rc7l2ozegds5xksy6qhrjin/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-n7su7jf54rc7l2ozegds5xksy6qhrjin/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-n7su7jf54rc7l2ozegds5xksy6qhrjin/bin/gfortran - skipped
-- Checking whether /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-n7su7jf54rc7l2ozegds5xksy6qhrjin/bin/gfortran supports Fortran 90
-- Checking whether /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen/gcc-8.3.1/gcc-10.2.0-n7su7jf54rc7l2ozegds5xksy6qhrjin/bin/gfortran supports Fortran 90 - yes
-- Found Git: /usr/bin/git (found version "2.27.0") 
-- Git description: v2022.05.05-19-g105d802b4
-- Could NOT find Python (missing: Interpreter) 
-- Found Yapf: /home/geoffrey/apps/py_env/bin/yapf (found version "0.29.0") 
-- yapf found at: /home/geoffrey/apps/py_env/bin/yapf
-- yapf version: 0.29.0
CMake Error at /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen2/gcc-10.2.0/cmake-3.18.2-nvinghkj7vuxvwvpkum5sdw5dv6ijrhj/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find Python (missing: Interpreter)
Call Stack (most recent call first):
  /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen2/gcc-10.2.0/cmake-3.18.2-nvinghkj7vuxvwvpkum5sdw5dv6ijrhj/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /cm/shared/apps/spack/cpu/opt/spack/linux-centos8-zen2/gcc-10.2.0/cmake-3.18.2-nvinghkj7vuxvwvpkum5sdw5dv6ijrhj/share/cmake-3.18/Modules/FindPython.cmake:436 (find_package_handle_standard_args)
  cmake/SetupGitHooks.cmake:22 (find_package)
  CMakeLists.txt:72 (include)


-- Configuring incomplete, errors occurred!
See also "/expanse/lustre/projects/csd275/geoffrey/build_expansesupport/CMakeFiles/CMakeOutput.log".
```

**Comment by nilsdeppe** [comment]
Expanse is down for maintenance right now so I can't re-test. I might have copied something from my expanse repo to my local machine incorrectly, so I'll double check. Thanks!

**Comment by nilsdeppe** [comment]
I can't reproduce @geoffrey4444's issue. This branch works for me as-is. If I remember correctly it also works for @knelli2.

@kidder can you please give this a try?

Rebased on develop

**Review by kidder — APPROVED** [review]

**Pull request merged**